test/e2e-olm/suite_test.go

```diff
@@ -0,0 +1,108 @@
+package e2e
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/reporters"
+	. "github.com/onsi/gomega"
+	consolev1 "github.com/openshift/api/console/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	apps "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	loggingv1 "github.com/openshift/elasticsearch-operator/apis/logging/v1"
+	// controllers "github.com/openshift/elasticsearch-operator/controllers/logging"
+	// +kubebuilder:scaffold:imports
+)
+
+const (
+	TestOperatorNamespaceEnv = "TEST_OPERATOR_NAMESPACE"
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+var (
+	cfg               *rest.Config
+	k8sClient         client.Client
+	testEnv           *envtest.Environment
+	operatorNamespace string
+	projectRootDir    string
+)
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	projectRootDir = getProjectRootPath("elasticsearch-operator")
+
+	junitReporter := reporters.NewJUnitReporter(filepath.Join(projectRootDir, "junit.xml"))
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Controller Suite",
+		[]Reporter{printer.NewlineReporter{}, junitReporter})
+}
+
+var _ = BeforeSuite(func(done Done) {
+	logf.SetLogger(zap.LoggerTo(GinkgoWriter, true))
+
+	By("bootstrapping test environment")
+	ok := false
```

Comment:
What is the purpose of the custom-enforced variable defaulting here?

Reply:
This is to avoid shadowing the global variable `operatorNamespace` at line 61.